x/tunnel/types/msgs.go

```diff
@@ -77,6 +77,19 @@ func NewMsgCreateIBCTunnel(
 	return m, nil
 }
 
+// NewMsgCreateIBCHookTunnel creates a new MsgCreateTunnel instance for IBC hook tunnel.
+func NewMsgCreateIBCHookTunnel(
+	signalDeviations []SignalDeviation,
+	interval uint64,
+	channelID string,
+	destinationContractAddress string,
+	initialDeposit sdk.Coins,
+	creator string,
+) (*MsgCreateTunnel, error) {
+	r := NewIBCHookRoute(channelID, destinationContractAddress)
+	return NewMsgCreateTunnel(signalDeviations, interval, r, initialDeposit, creator)
+}
```

Comment:
if you want to use this style (return calling), we do it for other routes here as well.